interface/patient_file/summary/dashboard_react/app/_devapp/components/Dashboard.js

```diff
@@ -0,0 +1,173 @@
+
+    }
+
+    /**
+     * Fetch current pid and activity component on component did mount
+     */
+    componentDidMount() {
+        let patientId = helpers.getPatientId();
+
+        this.setState({
+            patientId: patientId
+        });
+
+        //TEMP SOLUTION - create pid session
+        agent.PatientDataAgent.setGlobalPatientId(patientId);
+
+        this.fetchData("dashboard").then(result => {
+
+                let list = agent.Lists.getList("dashboard");
+                // console.log(list)
+                this.setState({'gadgets': list}, () => {
+                    this.updateGadgets();
+                });
+            }
+
+        );
+
+    }
+
+
+    /**
+     * fetch list from database using 'agent.Lists' and result boolean when list was saved in the agent.Lists
+     * @param listName
+     * @returns {Promise.<boolean>}
+     */
+    fetchData = async (listName) => {
+
+        var result = await agent.Lists.fetchList(listName);
+
+        return result ? true : false;
+    }
+
+    /**
+     * Run after the gadgets were returned from database.
+     * Update every state of section with your gadgets
+     */
+    updateGadgets () {
+        this.setState({
+            header:this.getElement('header'),
+            left:this.getElement('left'),
+            right:this.getElement('right')
+        })
+    }
+
+    /**
+     * return array of components that loaded using lazy load components
+     * @param id of section position - header/right/left
+     * @returns {Array}
+     */
+    getElement (id){
+
+        try {
+
+            let items = [];
+           // let elementObject = this.state.gadgets;
+
+            var result = [];
+
+            this.state.gadgets.map((v,i) => {
+                let item = v;
+
+                  if(item.notes.id == id && item.activity != 0) {
+                      // debugger;
+                      // console.log('-------');
+                      let element = item.notes.element;
+                      let title = item.title;
+                      let element_component = item.notes.element_component;
+                      let children = item.notes.children;
+                      let settings = item.notes.settings ? item.notes.settings : null;
+                      items.push(<LazyLoadModule key={element_component + i}
+                                                 resolve={() => import(/* webpackMode: "eager" */"./"+ element_component)}
+                                                 element={element}
+                                                 element_title={title}
+                                                 pid={this.state.patientId}
+                                                 children={children}
+                                                 settings={settings}
+
+                      />);
+                  }
+            })
+
+            return items;
+        }
+        catch (e)
+            {
+                console.log(e);
```

Comment:
Any thoughts on how you plan on notifying to the user that a component / module failed to load?  Or do you think this will only happen for developers?